openhtf/util/measurements.py

```diff
@@ -126,14 +127,23 @@ def Doc(self, docstring):
     self.docstring = docstring
     return self
 
-  def WithUnits(self, units):
+  def _maybe_make_unit_desc(self, unit_desc):
+    """Return the UnitDescriptor or convert a string to one."""
+    if isinstance(unit_desc, str):
+      unit_desc = units.Unit(unit_desc)
+    if not (isinstance(unit_desc, units.UnitDescriptor) or unit_desc is None):
+      raise TypeError('Invalid units for measurement %s: %s' % (self.name,
+                                                                unit_desc))
+    return unit_desc
+
+  def WithUnits(self, unit_desc):
     """Declare the units for this Measurement, returns self for chaining."""
-    self.units = units
+    self.units = self._maybe_make_unit_desc(unit_desc)
     return self
 
   def WithDimensions(self, *dimensions):
     """Declare dimensions for this Measurement, returns self for chaining."""
-    self.dimensions = dimensions
+    self.dimensions = [self._maybe_make_unit_desc(dim) for dim in dimensions]
```

Comment:
interestingly, dimensions used to be a tuple (`*args` is a tuple), now it's a list.  shouldn't matter, but i don't see any reason for it to be a list (it never gets modified), so `tuple(...)` might be better.